include/cpp-statsd-client/UDPSender.hpp

```diff
@@ -255,6 +263,17 @@ inline bool UDPSender::initialized() const noexcept {
     return m_socket != k_invalidFd;
 }
 
+inline void UDPSender::flush() noexcept {
+    // We aquire a lock but only if we actually need to (ie there is a thread also accessing the queue)
+    auto batchingLock =
+        m_batchingThread.joinable() ? std::unique_lock<std::mutex>(m_batchingMutex) : std::unique_lock<std::mutex>();
```

Comment:
technically, even if you are running with a background thread you could call `flush` and it would still work but you need to  lock first to do that. another option would be to just return if you are using the thread to send to the daemon. @vthiery do you have a preference? basically the other option would be to change this to:

```suggestion
    // The background thread takes care of sending to the daemon
    if (m_batchingThread.joinable()) {
        return;
    }
```

Reply:
This is fine as it is :+1: